app/models/king.rb

```diff
@@ -1,2 +1,5 @@
 class King < Piece
+  def valid_move?(x_end, y_end)
+    return true unless (x_end == horizontal_position && y_end == vertical_position) || (x_end - horizontal_position).abs > 1 || (y_end - horizontal_position).abs > 1
```

Comment:
since rubocop is complaining this is too long, we can refactor it something like:
```
did_not_move = x_end == horizontal_position && y_end == vertical_position
too_far = (x_end - horizontal_position).abs > 1 || (y_end - horizontal_position).abs > 1
return true unless did_not_move || too_far 
false # we still need to return false otherwise
```